File: reverseWordsInString.cpp
#include <iostream>
#include <stack>
using namespace std;

string reverseWords(string s)
{
    string result;
    stack<char> stack;
    for(int i=0;i<s.length();i++)
    {
        if(s[i]!=' ')
        {
            stack.push(s[i]);
        }
        else
        {
            while(!stack.empty())
            {
                result.push_back(stack.top());
                stack.pop();
            }
            result.push_back(' ');
        }
    }
    while(!stack.empty())
    {
        result.push_back(stack.top());
        stack.pop();
    }
    return result;
}

int main()
{
    string s;
    getline(cin,s);
    string result = reverseWords(s);
    cout<<result;
}

File: reverseWordsInString.py
class Solution:
    def reverseWords(self, str):
        stack = []
        result = ""
        
        for ch in str:
            if ch!=' ':
                stack.append(ch)
            else:
                while len(stack)!=0 :
                    result += stack.pop()
                result += ' '

        while len(stack)!=0 :
            result += stack.pop()
        
        return result

print(Solution().reverseWords("The cat in the hat"))
# ehT tac ni eht tah